daisy_rani_kesavan/week_01/day05/subway.html

```diff
@@ -0,0 +1,65 @@
+<html>
+<head>
+	Subway program
+</head>
+<body>
+<script type="text/javascript">
+
+var N=['Times Square','34th','28th','23rd','Union Square', '8th'];
+var L=['8th','6th','Union Square','3rd','1st'];
+var six=['Grand Central','33rd','28th','23rd','Union Square','Astor'];
+
+function rev(line1, stop1, line2, stop2){
+	var start= line1.indexOf(stop1);
+	var end= line2.indexOf(stop2);
+	var center=line2.indexOf('Union Square');
+	var temp=line2.slice(end, center);
+
+	if(end<center){
+      console.log("Your travel continue through the line2 : " + temp.reverse());
+    }
+    else{
+      console.log("You must travel through the stops on line line2 : " + line2.slice(center+1, line2.length));
+    }
+}
+
+function planTrip(line1, stop1, line2, stop2){
+
+	if(line1===line2){
+        var start= line1.indexOf(stop1);
+        console.log("You must travel through the stops on line : " + (line1).slice(start+1, line1.length));
+        }
+    
+    if(stop1==='Union Square'){
+    	console.log("You can travel directly to line2 from here");
+    	rev(line1, stop1, line2, stop2); 
+        }
+
+	if(stop1!=='Union Square'){
+        start= line1.indexOf(stop1);
+        console.log("You must travel through the stops on line line1 : " + (line1).slice(start+1, line1.length-1));
+        console.log("Change at Union Square");
+		rev(line1, stop1, line2, stop2);
+		 }
+
+ // 	if(line1===N || line1===six && line2===L){		
+ //        start= line1.indexOf(stop1);
+ //        console.log("You must travel through the stops on line lin1 : " + (line1).slice(start+1, line1.length-1));
+	// 	console.log("Change at Union Square");
+	// 	rev(line1, stop1, line2, stop2);
+ //    }
+
+
+  //   if(line1===L || line1===six && line2===N){		
+  //       start= line1.indexOf(stop1);
+  //       console.log("You must travel through the stops on line lin1 : " + (line1).slice(start+1, line1.length-1));
+		// console.log("Change at Union Square");
+		// rev(line1, stop1, line2, stop2);
+  //   }
+
+}
+
+planTrip(six, 'Astor', N, '34th');
```

Comment:
We haven't covered testing yet but these lines at the end where we call the function we wrote are similar to what we do when we write tests. I recommend trying several function calls, starting simple and getting trickier. Like....

```
planTrip(six, 'Astor', six, 'Astor'); // same line, same station
planTrip(six, '33rd', six, 'Astor'); // same line, travelling forwards
planTrip(six, 'Astor', six, '33rd'); // same line, travelling backwards
planTrip(six, 'Astor', N, '34th'); // change at union, travelling forwards
planTrip(N, '34th', six, 'Astor'); // change at union, travelling backwards
```